internal/services/batch/batch_pool.go

```diff
@@ -616,6 +656,190 @@ func ExpandBatchPoolStartTask(list []interface{}) (*batch.StartTask, error) {
+			}
+			result.ImageReference = imageReference
+		}
+	} else {
+		return nil, fmt.Errorf("storage_image_reference either is empty or contains parsing errors")
+	}
+
+	if containerConfiguration, err := ExpandBatchPoolContainerConfiguration(d.Get("container_configuration").([]interface{})); err == nil {
+		result.ContainerConfiguration = containerConfiguration
+	} else {
+		return nil, fmt.Errorf("container_configuration either is empty or contains parsing errors")
+	}
+
+	if dataDisk, diskErr := expandBatchPoolDataDisks(d.Get("data_disks").([]interface{})); diskErr == nil {
+		result.DataDisks = dataDisk
+	}
+
+	if diskEncryptionConfig, diskEncryptionErr := expandBatchPoolDiskEncryptionConfiguration(d.Get("disk_encryption_configuration").([]interface{})); diskEncryptionErr == nil {
+		result.DiskEncryptionConfiguration = diskEncryptionConfig
+	}
+
+	if extensions, extErr := expandBatchPoolExtensions(d.Get("extensions").([]interface{})); extErr == nil {
+		result.Extensions = extensions
+	}
+
+	if licenseType, ok := d.GetOk("license_type"); ok {
+		result.LicenseType = utils.String(licenseType.(string))
+	}
+
+	if nodeReplacementConfig, nodeRepCfgErr := expandBatchPoolNodeReplacementConfig(d.Get("node_placement_configuration").([]interface{})); nodeRepCfgErr == nil {
+		result.NodePlacementConfiguration = nodeReplacementConfig
+	}
+
+	if osDisk, osDiskErr := expandBatchPoolOSDisk(d.Get("os_disk_placement_setting")); osDiskErr == nil {
+		result.OsDisk = osDisk
+	}
+
+	if windowsConfiguration, windowsConfigErr := expandBatchPoolWindowsConfiguration(d.Get("windows_configuration").([]interface{})); windowsConfigErr == nil {
+		result.WindowsConfiguration = windowsConfiguration
+	}
+
+	return &result, nil
+}
+
+func expandBatchPoolOSDisk(ref interface{}) (*batch.OSDisk, error) {
+	if ref == nil {
+		return nil, fmt.Errorf("os_disk_placement_setting is empty")
+	}
+	item := ref.(string)
+	result := batch.OSDisk{
+		EphemeralOSDiskSettings: &batch.DiffDiskSettings{
+			Placement: batch.DiffDiskPlacement(item),
+		},
+	}
+	return &result, nil
+}
+
+func expandBatchPoolNodeReplacementConfig(list []interface{}) (*batch.NodePlacementConfiguration, error) {
+	if list == nil || len(list) == 0 || list[0] == nil {
+		return nil, fmt.Errorf("node_placement_configuration is empty")
+	}
+	item := list[0].(map[string]interface{})["policy"].(string)
+	result := batch.NodePlacementConfiguration{
+		Policy: batch.NodePlacementPolicyType(item),
+	}
+	return &result, nil
+}
+
+func expandBatchPoolWindowsConfiguration(list []interface{}) (*batch.WindowsConfiguration, error) {
+	if list == nil || len(list) == 0 || list[0] == nil {
+		return nil, fmt.Errorf("windows_configuration is empty")
+	}
+	item := list[0].(map[string]interface{})["enable_automatic_updates"].(bool)
+	result := batch.WindowsConfiguration{
+		EnableAutomaticUpdates: utils.Bool(item),
+	}
+	return &result, nil
+}
```

Comment:
and so frothj
```suggestion
func expandBatchPoolWindowsConfiguration(list []interface{}) (*batch.WindowsConfiguration, error) {
	if list == nil || len(list) == 0 || list[0] == nil {
		return nil, fmt.Errorf("windows_configuration is empty")
	}
	
	item := list[0].(map[string]interface{})["enable_automatic_updates"].(bool)
	return := &batch.WindowsConfiguration{
		EnableAutomaticUpdates: utils.Bool(item),
	}, nil
}
```

Reply:
Sure, I will check for other similar cases in code and update accordingly, thanks for your feedback.